### PR TITLE
New Difficulty Tests for ArrowGlacier

### PR DESCRIPTION
**Pull request by marioevz**
~~This PR contains updated tests from BasicTests to the new DifficultyTests category format.~~

~~Updated tests include: difficultyByzantium, difficultyConstantinople, difficultyFrontier, difficultyHomestead, difficultyEIP2384, difficultyEIP2384_random, difficultyEIP2384_random_to20M.~~

~~Currently, difficultyEIP2384_random and difficultyEIP2384_random_to20M fail due to ethereum/go-ethereum#23818.~~

**Update:** This PR now only contains ArrowGlacier tests.

New ArrowGlacier tests include:
- difficultyArrowGlacier: Combination checks with block numbers [10700000-15000000], timestamp deltas [1-90], uncles [0, 1]
- difficultyArrowGlacierFromLondon: London->ArrowGlacier gradual difficulty increase test: Test cases start at block 13460000 and difficulty=0x223a2d17232652 (taken from Mainnet), and gradually increase difficulty considering an average of 13s block time, and an average uncle frequency of 5%.
- difficultyArrowGlacierFromLondonAggressive: Same as previous test, but average uncle frequency is more aggressive at 15%.

All ArrowGlacier test cases are completing without issues.

### CODE REVIEW

**Comment by winsvega** [comment]
please. lets do one at a time. 
1 pr - remove old tests
2 pr - restore old test in new format 
3 pr - introduce new tests

**Review by winsvega — CHANGES_REQUESTED** [review]
plese lets divide and conquer

**Comment by marioevz** [comment]
I have added #976 and #977 to remove the BasicTests folder and update existing difficulty tests respectively.
I can rebase this branch to develop once those two PRs are merged.
Let me know if that is ok.

**Comment by marioevz** [comment]
This PR now only contains ArrowGlacier difficulty tests.
All other changes regarding old difficulty tests and removal of BasicTests files has been moved into different PRs.
@winsvega let me know if this addresses the requested changes.

File: src/DifficultyTestsFiller/dfArrowGlacier/difficultyArrowGlacierFromLondonCopier.json [hunk 1]
```diff
@@ -0,0 +1,2483 @@
+{
+    "difficultyArrowGlacierFromLondon": {
+        "_info": {
+            "comment": "London->ArrowGlacier gradual difficulty increase test: blocks 13460000 - 15000000, step 5000 blocks, uncle rate 5% "
```
Comment (winsvega) [inline src/DifficultyTestsFiller/dfArrowGlacier/difficultyArrowGlacierFromLondonCopier.json]: what is this test originate from?

File: src/DifficultyTestsFiller/dfArrowGlacier/difficultyArrowGlacierFromLondonAggressiveCopier.json [hunk 2]
```diff
@@ -0,0 +1,2483 @@
+{
+    "difficultyArrowGlacierFromLondonAggressive": {
+        "_info": {
+            "comment": "London->ArrowGlacier aggressive difficulty increase test: blocks 13460000 - 15000000, step 5000 blocks, uncle rate 15%"
```
Comment (winsvega) [inline src/DifficultyTestsFiller/dfArrowGlacier/difficultyArrowGlacierFromLondonAggressiveCopier.json]: source of the test?

File: src/DifficultyTestsFiller/dfArrowGlacier/difficultyArrowGlacierFromLondonCopier.json (hunk 1 above)
Comment (marioevz) [reply src/DifficultyTestsFiller/dfArrowGlacier/difficultyArrowGlacierFromLondonCopier.json]: I filled these test cases externally with a python script, I did not want the filler to make an exhaustive combination so I took a mainnet block as input and start point, and the script filled the steps of the test case where the difficulty gradually increases, to try to simulate the difficulty increase feedback loop in normal block production.

If this is not ok, I think another approach could be to add a filler with combinations on the block where block is supposed to happen.

**Pull request merged**